### PR TITLE
provider(sentry): Add experimental provider for Sentry

### PR DESCRIPTION
**Pull request by eseliger**
This PR adds another provider to the openctx repo: it takes a sentry issue URL and unfurls it into context for the AI assistant.

Test plan:

Tried it manually locally, and added some tests.

### CODE REVIEW

File: provider/sentry/README.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
+# Sentry issue context provider for OpenCtx
+
+This is a context provider for [OpenCtx](https://openctx.org) that fetches contents from a Sentty issue by just pasting the URL to it.
```
Comment (sqs) [inline provider/sentry/README.md:3]:
```suggestion
This is a context provider for [OpenCtx](https://openctx.org) that fetches contents from a Sentry issue by just pasting the URL to it.
```

Comment (eseliger) [reply provider/sentry/README.md]: oh no leaked my private sentty terminal emulator

File: provider/sentry/README.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+Then, add the token to your OpenCtx settings:
+
+```json
+"openctx.providers": {
+    // ...other providers...
+    "https://openctx.org/npm/@openctx/provider-sentry": {
+        "apiToken": "SENTRYTOKENHERE",
+        // or:
+        "apiTokenPath": "path/to/file/withsentrytoken"
```
Comment (tinvaan) [inline provider/sentry/README.md:39]: (nit) As discussed, might help to have optional settings for "project" & "organization". Solely relying on the sentry url for parsing the project and organization leaves us vulnerable to breaking changes on the sentry API.

Can be addressed separately in a future PR.

**Review by tinvaan — APPROVED** [review]
LGTM!

**Pull request merged**